soh/src/code/z_map_exp.c

```diff
@@ -470,37 +406,16 @@ void Map_InitRoomData(PlayState* play, s16 room) {
     osSyncPrintf("＊＊＊＊＊＊＊\n＊＊＊＊＊＊＊\nroom_no=%d (%d)(%d)\n＊＊＊＊＊＊＊\n＊＊＊＊＊＊＊\n", room,
                  mapIndex, play->sceneNum);
 
-    if (room >= 0) {
-        switch (play->sceneNum) {
-            case SCENE_DEKU_TREE:
-            case SCENE_DODONGOS_CAVERN:
-            case SCENE_JABU_JABU:
-            case SCENE_FOREST_TEMPLE:
-            case SCENE_FIRE_TEMPLE:
-            case SCENE_WATER_TEMPLE:
-            case SCENE_SPIRIT_TEMPLE:
-            case SCENE_SHADOW_TEMPLE:
-            case SCENE_BOTTOM_OF_THE_WELL:
-            case SCENE_ICE_CAVERN:
-            case SCENE_DEKU_TREE_BOSS:
-            case SCENE_DODONGOS_CAVERN_BOSS:
-            case SCENE_JABU_JABU_BOSS:
-            case SCENE_FOREST_TEMPLE_BOSS:
-            case SCENE_FIRE_TEMPLE_BOSS:
-            case SCENE_WATER_TEMPLE_BOSS:
-            case SCENE_SPIRIT_TEMPLE_BOSS:
-            case SCENE_SHADOW_TEMPLE_BOSS:
-                gSaveContext.sceneFlags[mapIndex].rooms |= gBitFlags[room];
-                osSyncPrintf("ＲＯＯＭ＿ＩＮＦ＝%d\n", gSaveContext.sceneFlags[mapIndex].rooms);
-                interfaceCtx->mapRoomNum = room;
-                interfaceCtx->unk_25A = mapIndex;
-                Map_SetPaletteData(play, room);
-                osSyncPrintf(VT_FGCOL(YELLOW));
-                osSyncPrintf("部屋部屋＝%d\n", room); // "Room Room = %d"
-                osSyncPrintf(VT_RST);
-                Map_InitData(play, room);
-                break;
-        }
+    if ((room >= 0) && (SceneDB_IsDungeon(play->sceneNum) || SceneDB_IsBoss(play->sceneNum))) {
```

Comment:
In this case, we don't have a SceneDB::Entry, we just have sceneNum, so we test both dungeon and boss.